verilog/tools/ls/lsp-parse-buffer.cc

```diff
@@ -67,7 +67,9 @@ BufferTrackerContainer::GetSubscriptionCallback() {
         if (txt) {
           const BufferTracker *tracker = Update(uri, *txt);
           // Now inform our listeners.
-          if (change_listener_) change_listener_(uri, *tracker);
+          for (const auto &change_listener : change_listeners_) {
+            change_listener(uri, *tracker);
+          }
         } else {
           Remove(uri);
```

Comment:
On called back with this callback, you store references to the text structure stored in tracker data.

But if the file gets removed, there is no change listener that informs about a URI being removed (the `Remove(uri)` below does not call any listeners)...

Which means in case the URI is deleted (someone closes the buffer in the editor)  you'll store a dangeling reference to a removed text buffer in `VerilogLanguageServer::UpdateEditedFileInProject()` -> ` SymbolTableHandler::UpdateFileContent()`.

I suggest to have the change listener to take a pointer, which can inform about a `nullptr` buffer (i.e. inform about a removed URI), which you then can also use to properly clean up references.

Also, longer term it is probably better to have `BufferTracker`own the `ParsedVerilogSourceFile` which we then pass to `VerilogProject::UpdateFileContents()`. Right now, due to the chain of change listeners, the ownership is quite murky and will result in issues.